apps/reference/docs/guides/auth/server-side-rendering.mdx

```diff
@@ -0,0 +1,190 @@
+
+
+```typescript
+const refreshToken = req.cookies['my-refresh-token'];
+
+if (refreshToken) {
+  await supabase.auth.setSession(refreshToken);
+} else {
+  // make sure you handle this case!
+  throw new Error("User is not authenticated.")
+}
+
+// returns user information
+await supabase.auth.getUser();
+```
+
+Use `setSession(refreshToken)` instead of `getUser(accessToken)` as refresh
+tokens are long-lived credentials that provide user information even if the user
+has not used your app in a long time.
+
+Even though refresh tokens are long-lived, they is no guarantee that a user
+has an active session. They may have logged out and your application failed to
+remove the `my-refresh-token` cookie, or some other failure occurred that left
+a stale refresh token in the browser.
+
+A good practice is to handle unauthorized errors by deferring to render the page
+in the browser instead of in the server. Some user information is contained in the
+access token though, so in certain cases, you may be able to use this potentially
+stale information to render a page.
+
+## Frequently Asked Questions
+
+### How do I make the cookies `HttpOnly`?
+
+This is not necessary. Both the access token and refresh token are designed to
+be passed around to different components in your application. The browser-based
+side of your application needs access to the refresh token to properly maintain
+a browser session anyway.
+
+### My server is getting invalid refresh token errors. What's going on?
+
+It is likely that the refresh token sent from the browser to your server is
+stale. Make sure the `onAuthStateChange` listener callback is free of bugs and
+is registered relatively early in your application's lifetime.
+
+When you receive this error on the server-side, try to defer
+rendering to the browser where the client library can access an up-to-date
+refresh token and present the user with a better experience.
+
+### Should I set a shorter `Max-Age` parameter on the cookies?
+
+The `Max-Age` or `Expires` cookie parameters only control whether the browser
+sends the value to the server. Since a refresh token represents the
+long-lived authentication session of the user on that browser, setting a short
+`Max-Age` or `Expires` parameter on the cookies only results in a degraded
+user experience.
+
+The only way to ensure that a user has logged out or their session has ended
+is to get the user's details with `getUser()`.
+
+### What should I use for the `SameSite` property?
+
+Make sure you [understand the behavior of the parameter in different
+situations](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite),
+as your users can be experiencing a degraded user experience.
+
+A good default is to use `Lax` which will send cookies when users are
+navigating to your site. Cookies typically require the `Secure` attribute,
+which will only send them over HTTPS. However, this can sometimes be a problem
+when developing on `localhost`.
```

Comment:
```suggestion
Make sure you [understand the behavior of the property in different
situations](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite)
as some properties can degrade the user experience.

A good default is to use `Lax` which sends cookies when users are
navigating to your site. Cookies typically require the `Secure` attribute,
which only sends them over HTTPS. However, this can be a problem when
developing on `localhost`.
```